File: src/codility/java/stack_queue/Brackets.java
package codility.java.stack_queue;

import java.util.Stack;

public class Brackets {

    public static void main(String[] args) {

    }

    public int solution(String S) {

        Stack<Character> stack = new Stack<>();
        for (char c : S.toCharArray()) {
            if (c == '[' || c == '{' || c== '(') {
                stack.push(c);
            } else {
                if (stack.isEmpty()) {
                    return 0;
                }
                char top = stack.pop();

                if (top == '(' && c != ')') {
                    return 0;
                }
                else if (top == '{' && c != '}') {
                    return 0;
                }
                else if (top == '[' && c != ']') {
                    return 0;
                }
            }

        }
        return stack.isEmpty()? 1: 0;
    }
}

File: src/codility/kotlin/Brackets.kt
package codility.kotlin

import java.util.*

class Brackets {

    fun solution(S: String): Int {
        // write your code in Kotlin

        val stack = Stack<Char>()

        val map = mapOf<Char, Char>('{' to '}', '(' to ')', '[' to ']')

        for (c in S) {
            if (c == '{' || c == '[' || c =='(') {
                stack.add(c)
            } else {
                if (stack.isEmpty()) {
                    return 0
                }

                val top = stack.pop()
                if (map[top] != c) {
                    return 0
                }

            }
        }

        return if (stack.empty()) 1 else 0
    }

}